simde/x86/avx512/range.h

```diff
@@ -0,0 +1,118 @@
+#if !defined(SIMDE_X86_AVX512_RANGE_H)
+#define SIMDE_X86_AVX512_RANGE_H
+
+#include "types.h"
+#include "max.h"
+#include "min.h"
+#include "set1.h"
+#include "and.h"
+#include "or.h"
+#include "copysign.h"
+#include "abs.h"
+#include "setzero.h"
+#include "cmp.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+SIMDE_FUNCTION_ATTRIBUTES
+simde__m512d
+simde_mm512_range_pd (simde__m512d a, simde__m512d b, int imm8)
+    SIMDE_REQUIRE_CONSTANT_RANGE(imm8, 0, 15) {
+  #if defined(SIMDE_X86_AVX512DQ_NATIVE)
+    simde__m512d r;
+    SIMDE_CONSTIFY_16_(_mm512_range_pd, r, (HEDLEY_UNREACHABLE(), simde_mm512_setzero_pd ()), imm8, a, b);
+    return r;
+  #else
+    simde__m512d r;
+
+    if (imm8 & 1) {
```

Comment:
This would be a lot easier to follow using a switch.  Not only that, but we want to avoid branching, and compilers often have an easier time optimizing a switch to do that.

```c
switch (imm8) {
  case 1:
    ...
    break;
  case 2:
    ...
    break;
}
```